packages/zoe/README.md

```diff
@@ -124,3 +132,25 @@ harden(prepare);
 ```
 
 For an example contract upgrade, see the test at https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/test/swingsetTests/upgradeCoveredCall/test-coveredCall-service-upgrade.js .
+
+### Durability
+
+The contract must retain in durable storage anything that must persist between incarnations. All other state will be lost.
+
+### Kinds
+
+The contract defines the kinds that are held in durable storage. Thus the function calls that define the kinds must be run before the objects are deserialized from durable storage.
+
+# Crank
+
+After the first incarnation, `prepare` must return in one crank. Therefore any
+remote calls must only be during the initial incaranation. During subsequent
+upgrades, `prepare` should be using a value that it stashed away in the baggage.
+The `provideAll` function in contract support is designed to help.
+
+The reason is that all vats must be able to finish their upgrade without
+contacting other vats. There might be messages queued inbound to the vat being
+upgraded, and the kernel safely deliver those messages until the upgrade is
+complete. The kernel can't tell which external messages are needed for upgrade,
```

Comment:
"complete. Any outbound messages the contracts sends during upgrade will eventually provoke an inbound `notify` (result promise resolution) message. These result messages will be queued *behind* any pre-existing inbound messages, and the kernel cannot tell which inbound messages are needed for upgrade, as opposed to existing work that must be delayed until upgrade is finished.

So, the rule is that `buildRootObject()` (and thus `prepare()`) must be standalone, and complete without waiting for off-vat responses."